app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/offline/CachedDatabaseService.kt

```diff
@@ -48,12 +61,23 @@ class CachedDatabaseManagement internal constructor(
         return Converters.fromPicture(picture, uri)
     }
 
-    private fun removeFromCache(pictureId: Id) {
+    private suspend fun removeFromCache(pictureId: Id) {
         cachedPictures.remove(pictureId)
         try {
             cache.deletePicture(pictureId)
         } catch (e: DatabaseService.NotFoundException) {
             // Do nothing since it means that the cache already removed it itself
         }
     }
+
+    private suspend fun getRepresentativePictureFromCache(pictureId: Id): CategorizedPicture? {
```

Comment:
I don't really understand what this function does and why it needs to be this convoluted. It might be good to briefly explain what problem it's supposed to solve.

Reply:
This function is called if the representative picture has already been cached for this category (line 40) and then tries to get it from cache or reloads it if it is no longer in there.
This makes me think that having a list which keeps track of representative pictures is a bad idea since those could change without it being updated. I will remove this list and always retrieve the picture id of the representative picture from firestore instead.